### PR TITLE
feat(cli-mock): add section for searchable mocking

### PR DESCRIPTION
**Pull request by phani-srikar**
_Issue #, if available:_

_Description of changes:_
Document the support for mocking the `@searchable` GraphQL directive.
**DO NOT MERGE** until CLI >=`10.5.1` is GA.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/pages/cli/usage/mock.mdx [hunk 1]
```diff
@@ -56,10 +58,23 @@ The environment variables that are listed below in [Function mock environment va
 
 In addition, you can use a `.env` file within the function directory (ie. `<project root>/amplify/backend/function/<function name>/.env`) to override any environment variables for local mocking.
 
+### Mocking the searchable models
+If you use `@searchable` directives on `@model` types in your GraphQL schema, then you can test the search GraphQL queries that are generated for you locally via `amplify mock` or `amplify mock api`.
+To learn more about the search queries, refer [this guide](/cli/graphql/search-and-result-aggregations/).
```
Comment (phani-srikar) [inline src/pages/cli/usage/mock.mdx:63]: Do you think we need an example GraphQL search query here? Different types of search queries are present in the guide I linked here. Would that suffice?

File: src/pages/cli/usage/mock.mdx [hunk 2]
```diff
@@ -45,7 +45,9 @@ When defining a schema you can use directives from the GraphQL Transformer in lo
 - [@primaryKey, @index](/cli/graphql/data-modeling)
 - [@hasOne, @hasMany, @belongsTo, @manyToMany](/cli/graphql/data-modeling)
 - [@function](/cli/graphql/custom-business-logic#lambda-function-resolver)
+- [@searchable](/cli/graphql/search-and-result-aggregations/)
 
+### Mocking the lambda triggers on models
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:50]:
```suggestion
### Mocking the Lambda triggers on @model types
```

File: src/pages/cli/usage/mock.mdx [hunk 3]
```diff
@@ -56,10 +58,23 @@ The environment variables that are listed below in [Function mock environment va
 
 In addition, you can use a `.env` file within the function directory (ie. `<project root>/amplify/backend/function/<function name>/.env`) to override any environment variables for local mocking.
 
+### Mocking the searchable models
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:61]:
```suggestion
### Mocking @model types with @searchable
```

File: src/pages/cli/usage/mock.mdx [hunk 4]
```diff
@@ -56,10 +58,23 @@ The environment variables that are listed below in [Function mock environment va
...
+If you use `@searchable` directives on `@model` types in your GraphQL schema, then you can test the search GraphQL queries that are generated for you locally via `amplify mock` or `amplify mock api`.
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:62]:
```suggestion
If you use the `@searchable` directive on `@model` types in your GraphQL schema, then you can test the search GraphQL queries that are generated for you locally via `amplify mock` or `amplify mock api`.
```

File: src/pages/cli/usage/mock.mdx (hunk 1 above)
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:63]:
```suggestion
To learn more about the search queries, refer to [this guide](/cli/graphql/search-and-result-aggregations/).
```

File: src/pages/cli/usage/mock.mdx [hunk 5]
```diff
@@ -56,10 +58,23 @@ The environment variables that are listed below in [Function mock environment va
...
+To learn more about the search queries, refer [this guide](/cli/graphql/search-and-result-aggregations/).
+
+We create the following artifacts when you mock the API with searchable models:
+- [Opensearch](https://opensearch.org/) version `1.3.0` is downloaded.
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:66]:
```suggestion
- [OpenSearch](https://opensearch.org/) version `1.3.0` is downloaded.
```

File: src/pages/cli/usage/mock.mdx [hunk 6]
```diff
@@ -56,10 +58,23 @@ The environment variables that are listed below in [Function mock environment va
...
+- A python lambda trigger is stored locally at `mock-api-resources/searchable/searchable-lambda-trigger`.
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:67]:
```suggestion
- A python Lambda trigger is stored locally at `mock-api-resources/searchable/searchable-lambda-trigger`. 
```

File: src/pages/cli/usage/mock.mdx [hunk 7]
```diff
@@ -56,10 +58,23 @@ The environment variables that are listed below in [Function mock environment va
...
 In addition, you can use a `.env` file within the function directory (ie. `<project root>/amplify/backend/function/<function name>/.env`) to override any environment variables for local mocking.
 
+### Mocking the searchable models
+If you use `@searchable` directives on `@model` types in your GraphQL schema, then you can test the search GraphQL queries that are generated for you locally via `amplify mock` or `amplify mock api`.
+To learn more about the search queries, refer [this guide](/cli/graphql/search-and-result-aggregations/).
+
+We create the following artifacts when you mock the API with searchable models:
+- [Opensearch](https://opensearch.org/) version `1.3.0` is downloaded.
+- A python lambda trigger is stored locally at `mock-api-resources/searchable/searchable-lambda-trigger`. 
+This lambda trigger is automatically invoked locally if you perform a CRUD operation on the searchable model types.
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:68]:
```suggestion
This Lambda trigger is automatically invoked locally if you perform a CRUD operation on the searchable model types. 
```

File: src/pages/cli/usage/mock.mdx [hunk 8]
```diff
@@ -56,10 +58,23 @@ The environment variables that are listed below in [Function mock environment va
...
 
+### Mocking the searchable models
+If you use `@searchable` directives on `@model` types in your GraphQL schema, then you can test the search GraphQL queries that are generated for you locally via `amplify mock` or `amplify mock api`.
+To learn more about the search queries, refer [this guide](/cli/graphql/search-and-result-aggregations/).
+
+We create the following artifacts when you mock the API with searchable models:
+- [Opensearch](https://opensearch.org/) version `1.3.0` is downloaded.
+- A python lambda trigger is stored locally at `mock-api-resources/searchable/searchable-lambda-trigger`. 
+This lambda trigger is automatically invoked locally if you perform a CRUD operation on the searchable model types. 
+It updates the data in the corresponding search index used by the locally running opensearch cluster.
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:69]:
```suggestion
It updates the data in the corresponding search index used by the local OpenSearch cluster.
```

File: src/pages/cli/usage/mock.mdx [hunk 9]
```diff
@@ -56,10 +58,23 @@ The environment variables that are listed below in [Function mock environment va
...
+- A data folder to store the opensearch indices is created at `mock-api-resources/searchable/searchable-data`.
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:70]:
```suggestion
- A data folder to store the OpenSearch indices is created at `mock-api-resources/searchable/searchable-data`.
```

**Review by renebrandel — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**